src/user/tag.ts

```diff
@@ -0,0 +1,86 @@
+import { log } from "../config"
+import { Accessory } from "../accessory"
+import { instanceToClass } from "clone-class"
+import { TagPayload } from "wechaty-puppet/dist/src/schemas/tag"
+
+export class Tag  extends Accessory {
+
+  /**
+   * @private
+   */
+  constructor (
+    public readonly id: string,
+  ) {
+    super()
+    log.silly('Tag', `constructor(${id})`)
+
+    // tslint:disable-next-line:variable-name
+    const MyClass = instanceToClass(this, Tag)
+
+    if (MyClass === Tag) {
+      throw new Error(
+        'Tag class can not be instanciated directly!'
+        + 'See: https://github.com/Chatie/wechaty/issues/1217',
+      )
+    }
+
+    if (!this.puppet) {
+      throw new Error('Tag class can not be instanciated without a puppet!')
+    }
+  }
+
+  public static async findAll<T extends typeof Tag> (
+    this  : T,
+  ): Promise<TagPayload []> {
+    log.verbose('Tag', 'findAll()')
+
+    try {
+      const tagList: TagPayload[] = await this.puppet.allTags()
+
+      return tagList
+    } catch (e) {
+      log.error('Tag', 'this.puppet.contactFindAll() rejected: %s', e.message)
+      return [] // fail safe
+    }
+  }
+
+  public static async find<T extends typeof Tag> (
+    this  : T,
+    query : string
+  ): Promise<TagPayload> {
+    log.verbose('Tag', 'find(%s)', JSON.stringify(query))
+
+    const tag = await this.puppet.searchTag(query)
+
+    return tag
+  }
+
+  /**
+   * show all labels across the login account
+   */
+  public async tagList (): Promise<TagPayload[]> {
+    log.verbose('Tag', 'tagList()')
+
+    try {
+      return this.puppet.tagList()
+    } catch (e) {
+      log.error('Tag', 'tagList() exception: %s', e.message)
+      throw new Error(`can not get tag list`)
+    }
+  }
+
+  /**
+   * create a new label
+   */
+  public async newTag (tag: string): Promise<string> {
```

Comment:
newTag -> create

I believe it will be better to make a call like `bot.Tag.create('myTag')` instead of `bot.Tag.newTag('myTag')`.

Reply:
Yes, you are right, `create` is better than `newTag`.